src/main/java/org/benetech/servicenet/service/impl/LocationMatchServiceImpl.java

```diff
@@ -35,6 +69,16 @@ public LocationMatch saveOrUpdate(LocationMatch locationMatch) {
 
     @Override
     public void delete(UUID locationId, UUID matchingLocationId) {
-        locationMatchRepository.deleteByLocationIdAndMatchingLocationId(locationId, matchingLocationId);
+        Optional<LocationMatch> existingMatch = locationMatchRepository.findByLocationIdAndMatchingLocationId(
```

Comment:
Perhaps we should add some constraint to ensure that there can be only one location/service match for given ids

Reply:
what would happen if we tried saving the same match concurrently?